### PR TITLE
Fix windows build

### PR DESCRIPTION
**Pull request by SylvainJuge**
The `-Dmaven.gitcommitid.skip=...`  maven CLI argument gets ignored when we also add `skipViaCommandLine` in the plugin configuration.

Found the underlying issue details in the source code : https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/315

This configuration was initially added as an attempt to keep the build compatible with git-worktrees (which I'm personally heavily reliant on). The proper (and seamless) fix is to activate a maven profile to switch to `git` native client only when required instead of skipping the whole `git-commit-id` plugin through configuration.

### CODE REVIEW

File: apm-agent-plugins/apm-logging-plugin/apm-log4j2-plugin/src/test/java/co/elastic/apm/agent/log4j2/Log4j2InstrumentationTest.java [hunk 1]
```diff
@@ -92,7 +96,10 @@ public Log4j2LoggerFacade() {
         @Override
         public void open() {
             log4j2Logger = LogManager.getLogger("Test-File-Logger");
-            ((org.apache.logging.log4j.core.Logger) log4j2Logger).getContext().setConfigLocation(configLocation);
+            assertThat(log4j2Logger).isInstanceOf(org.apache.logging.log4j.core.Logger.class);
+
+            LoggerContext loggerContext = ((org.apache.logging.log4j.core.Logger) log4j2Logger).getContext();
+            loggerContext.setConfigLocation(configLocation);
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-logging-plugin/apm-log4j2-plugin/src/test/java/co/elastic/apm/agent/log4j2/Log4j2InstrumentationTest.java:102]: [for reviewer] this does not help fixing the tests, but would help to identify test flakyness on windows

File: apm-agent-plugins/apm-logging-plugin/apm-log4j2-plugin/src/test/java/co/elastic/apm/agent/log4j2/Log4j2InstrumentationTest.java [hunk 2]
```diff
@@ -102,7 +109,15 @@ public void close() {
...
-            return ((RandomAccessFileAppender) ((org.apache.logging.log4j.core.Logger) log4j2Logger).getAppenders().get("FILE")).getFileName();
+            assertThat(log4j2Logger)
+                .isInstanceOf(org.apache.logging.log4j.core.Logger.class);
+            org.apache.logging.log4j.core.Logger logger = (org.apache.logging.log4j.core.Logger) log4j2Logger;
+
+            assertThat(logger.getAppenders()).containsKey("FILE");
+            Appender fileAppender = logger.getAppenders().get("FILE");
+            assertThat(fileAppender).isInstanceOf(RandomAccessFileAppender.class);
+
+            return ((RandomAccessFileAppender) fileAppender).getFileName();
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-logging-plugin/apm-log4j2-plugin/src/test/java/co/elastic/apm/agent/log4j2/Log4j2InstrumentationTest.java:120]: [for reviewer] this does not help fixing the tests, but would help to identify test flakyness on windows

**Comment by SylvainJuge** [comment]
Making it as ready for review, even if Windows tests still fail, at least we are making some progress in the right direction and fix the `git` command issue.

**Review by jackshirazi — APPROVED** [review]
Build running on Windows is succeeding for me

**Pull request merged**